### ISSUE TITLE
File-Links with german Umlauts don't work

### ISSUE THREAD

**Issue by Tronnyi** [laurent22/joplin#4043]
[Präsentation](file:///T:/...Präsentation.doc) <- nothing happens when clicking this link
[Präsentation](file:///T:/...Praesentation.doc) <- works

Drag and drop + Alt of a file with Umlauts creates a faulty link, wich also doesn't work:
[Präsentation](file:///T:/...pr%C3%A4sentation-doc)
changing the link manually by changing the symbols to an "ä" doesn't solve the issue.

## Environment

Joplin version:  1.2.6 (prod, win32)
Platform:  Win10

**Comment by tessus** [comment]
I added a fix a while back that encodes the link properly, so I'm not sure why this does not work anymore. Maybe it got lost during the refactoring.
Or maybe there were changes in how Electron handles URL en/decoding. It could also be an issue with UTF8.

The encoding for ascii of `ä` is `%E4`. But in UTF8 it is `%C3%A4`.

**Comment by laurent22** [comment]
> Präsentation <- nothing happens when clicking this link

That's not a link. For us to replicate the issue, we need the actual Markdown code.

**Comment by th-ros** [comment]
I tried to reproduce some cases:

Joplin v1.4.12 (pre-release), Windows Portable

```
**Tests without umlaut:**
<h:/test.txt> not rendered as link
<file://h:/test.txt> ok
[local file](h:/test.txt) ok
[local file](file://h:/test.txt) ok

**Test with umlaut:**
<h:/test_with_umlaut_ä.txt> not rendered as link
<file://h:/test_with_umlaut_ä.txt> not working
<file://h:/test_with_umlaut_ae.txt> not working (ä changed with ae)
<file://h:/test_with_umlaut_%E4.txt> not working (ä changed with %E4 for ASCII encoding)
<file://h:/test_with_umlaut_%C3%A4.txt> not working (ä changed with %C3%A4 for UTF-8 encoding)
[local file with umlaut without file://...](h:/test_with_umlaut_ä.txt) not working
[local file with umlaut with file://...](file://h:/test_with_umlaut_ä.txt) not working

<https://de.wikipedia.org/wiki/Thomas_Müller> ok
[Wikipedia Thomas Müller](https://de.wikipedia.org/wiki/Thomas_Müller) ok

[test_with_umlaut_ä.txt](:/9d72c4546ccf4c8d8b6d9ac7bf032215) ok (file added to joplin via drag and drop)
```

**Comment by th-ros** [comment]
I tested the markdown code from my comment above with the actual version 1.5.10 (pre-release) and the result is the same. So I think it's not fixed (if it's a bug).

**Comment by laurent22** [comment]
I can't replicate this by copying / pasting text from the examples because it most likely has something to do with character encoding. And if I copy from here, it's UTF-8 so it works.

To replicate this, we would need a complete test case:

- One file with umlaut
- One JEX file that links to that other file

All that **in a ZIP archive** so that character encoding is preserved.

**Comment by th-ros** [comment]
Maybe this helps.
[test_files_for_joplin_issue_#4043.zip](https://github.com/laurent22/joplin/files/5752143/test_files_for_joplin_issue_.4043.zip)

**Comment by rpld** [comment]
I ran into a similar problem, not certain if the same issue.
I'm also not sure if it's an issue with Joplin, might have been the fault of OneNote I imported from.

This link works fine: https://github.com/opengapps/aosp_build.git
This link I copied from Joplin is broken: https://github.com/opengapps/aosp_build.git﻿

I can copy and paste these links and the problem remains.
They look identical, but I think the broken one adds a "%EF%BB%BF" at the end.
v1.7.10
Hope this helps, thanks.
